web/modules/custom/dpl_event/src/Access/EventSeriesTabAccessCheck.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\dpl_event\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Provides an access checker for the event series tab.
+ *
+ * This access check ensures that the tab for editing event instances
+ * is only available when there are more than one instance in the series.
+ */
+class EventSeriesTabAccessCheck implements AccessInterface {
+
+  /**
+   * The entity type manager service.
+   */
+  public function __construct(
+    private EntityTypeManagerInterface $entityTypeManager
+  ) {}
+
+  /**
+   * Access callback for the event series edit instances tab.
+   *
+   * Checks if there are more than one instance in the event series and grants
+   * or denies access based on this condition.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route
+   *   The route match interface to get route parameters.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *   The access result, allowing access if there are more than one instance,
+   *   otherwise forbidding it.
+   */
+  public function access(RouteMatchInterface $route): AccessResult {
+    $eventSeriesId = $route->getParameter('eventseries');
+    if (!$eventSeriesId) {
+      return AccessResult::forbidden();
+    }
+
+    $eventSeriesEntity = $this->entityTypeManager->getStorage('eventseries')->load($eventSeriesId);
+    if (!$eventSeriesEntity || !method_exists($eventSeriesEntity, 'getInstanceCount')) {
```

Comment:
Please check whether the loaded object is an instance of the expected class instead of checking whether the method exists.

This is a more explict way of ensuring we have the expected value. 

In fact we know that the entity storage for event series will always return an `EventSeries` object. [We can tell that to PHPStan](https://github.com/mglaman/phpstan-drupal?tab=readme-ov-file#entity-storage-mappings) but PHPStorm will not know anyway so I guess having an explicit check here is good for now.